### PR TITLE
Enhance overlapping multi-avatar display

### LINKED ISSUE #6019: Enhance multiple avatar display
For the Android Deck app, @stefan-niedermann added a very nice 2dp border as per https://github.com/stefan-niedermann/nextcloud-deck/issues/433

Would be very nice to use this in the file list as well, and possibly in other places where we show multiple avatars:

> grafik
> grafik

@stefan-niedermann also said:

> In theory this is reusable.
> 
> @tobiasKaminsky if you want to reuse this component, i will put some heart into it and extract it into a own module package.
> 
> But beware, the border is a hard color value which has to be set depending on where you are using the OverlappingAvatarsLayout - because in the sample case the background of the material card uses a transparent color, and if use the exact same color it will multiply and the the border-color will be darker than the background of the cards.
> 
> TL;DR: Contact me in Telegram if you want to reuse it.

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Resolves #6019

@stefan-niedermann I took the liberty to add you as an author to the implementation since I borrowed some of your code like mentioned in the issue #6019

Signed-off-by: Andy Scherzinger <info@andy-scherzinger.de>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [x] Tests written, or not not needed

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
:+1: great work.

Looking forward to screenshot tests :+1:

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I vote to merge this one too since everything is fine except screenshot updates, which isn't related to this PR
